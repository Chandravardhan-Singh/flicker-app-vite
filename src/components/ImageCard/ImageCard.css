.imageCard {
    width:90%;
    height: auto;
}
.media-container{
    overflow: hidden;
    background-color: #f00;
    margin:10px auto;
}
.imageCard:hover {
    scale: 1.3;
    animation-name: scaling;
    animation-duration: 0.5s;
}

@media screen and (max-width:480px){
    .imageCard{
        height: 100px;
    }
}

@keyframes scaling {
    from {scale:1}
    to {scale: 1.3;}
  }
