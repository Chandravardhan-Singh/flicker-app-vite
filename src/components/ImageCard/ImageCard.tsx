import React, { FC } from "react";
import { IImage } from "../../redux/feature/products/flickerSlice";
import "./ImageCard.css";
import { Card, CardContent, CardMedia, Typography } from "@mui/material";
import { Link } from "react-router-dom";
import { RoutePaths } from "../../routes/route-path.enum";
import { IMAGE_URL } from "../../utils";

interface IImageProp {
  image: IImage;
}

const ImageCard: FC<IImageProp> = (props) => {
  return (
    <Card style={{ marginTop: 10 }} data-testid="todo-1">
      <div className="media-container">
        <Link to={`${RoutePaths.ImageDetails}/${props.image?.id}`}>
          <CardMedia
            component={"img"}
            image={`${IMAGE_URL}/${props.image?.server}/${props.image?.id}_${props.image?.secret}_q.jpg`}
            className="imageCard"
          />
        </Link>
      </div>
      <CardContent>
        <Typography color={"#000"} variant="caption">
          {props.image?.title}
        </Typography>
      </CardContent>
    </Card>
  );
};

export default ImageCard;
