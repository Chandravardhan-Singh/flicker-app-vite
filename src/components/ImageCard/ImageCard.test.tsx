import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import ImageCard from "./ImageCard";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
const data = {
  id: "52744629460",
  owner: "197296781@N03",
  secret: "6c4c4c4726",
  server: "65535",
  farm: 66,
  title: ".:: ::.",
  ispublic: 1,
  isfriend: 0,
  isfamily: 0,
};

afterEach(() => {
  cleanup();
});

test("should render ImageCard", () => {
  render(<ImageCard image={data} />, {
    wrapper: BrowserRouter,
  });
  const imageElement = screen.getByTestId("todo-1");
  expect(imageElement).toBeInTheDocument();
  expect(imageElement).toHaveTextContent(data.title);
});

// snapshot
test("match snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <ImageCard image={data} />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
