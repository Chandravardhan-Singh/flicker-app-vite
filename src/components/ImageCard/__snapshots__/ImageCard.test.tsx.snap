// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  data-testid="todo-1"
  style={
    {
      "marginTop": 10,
    }
  }
>
  <div
    className="media-container"
  >
    <a
      href="/image/52744629460"
      onClick={[Function]}
    >
      <img
        className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img imageCard css-o69gx8-MuiCardMedia-root"
        src="https://live.staticflickr.com/65535/52744629460_6c4c4c4726_q.jpg"
      />
    </a>
  </div>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <span
      className="MuiTypography-root MuiTypography-caption css-sppdzw-MuiTypography-root"
    >
      .:: ::.
    </span>
  </div>
</div>
`;
